validations.go

```diff
@@ -74,6 +106,21 @@ type validateUpdateable interface {
 }
 
 func (m *Model) validateUpdate(c *Connection) (*validate.Errors, error) {
+	v := reflect.Indirect(reflect.ValueOf(m.Value))
```

Comment:
I feel like a lot of this logic is repeated. would make sense to break it out into it's own method?